cmd/libs/go2idl/deepcopy-gen/generators/deepcopy.go

```diff
@@ -157,7 +154,26 @@ func (g *genDeepCopy) copyableWithinPackage(t *types.Type) bool {
 		return false
 	}
 	// TODO: Consider generating functions for other kinds too.
-	return t.Kind == types.Struct
+	if t.Kind != types.Struct {
+		return false
+	}
+	// TODO: This should be removed once we start generating public DeepCopy
+	// functions per directory.
+	if t.Name.Package != g.targetPackage {
+		// We won't be able to access private fields.
+		// Thus, this type cannot have private fields.
+		for _, member := range t.Members {
+			if strings.ToLower(member.Name[:1]) == member.Name[:1] {
+				return false
+			}
+			// TODO: This is a temporary hack, to make avoid generating function
+			// for conversion.Equalities. We should get rid of it.
```

Comment:
I think lavalamp has reviewed this part, but out of curiosity, why are the functions for conversion.Equalities related to if a member is embedded or not?